src/Development/Shake/Command.hs

```diff
@@ -561,6 +561,15 @@ type a :-> t = a
 cmd :: CmdArguments args => args :-> Action r
 cmd = cmdArguments []
 
+class Unit a
+instance {-# OVERLAPPING #-} Unit b => Unit (a -> b)
+instance {-# OVERLAPPABLE #-} a ~ () => Unit (m a)
```

Comment:
Drat -- you're right. New commits split Unit into its own module that imports yet another module based on which GHC version is available. [apparently GHC doesn't use cpp to parse language pragmas so something like
```
{-# LANGUAGE CPP #-}
#if foo
{-# LANGUAGE OverlappingInstances
#endif
...
```
didn't work]